scripts/tests/z-wave-stack-binaries-test.sh

```diff
@@ -0,0 +1,508 @@
+        [ -z $devel ] || { cat $mqtt_sub_log ; sleep 20 ; }
+        tail -n +$count "$mqtt_sub_log"  | head -n1 | grep -F "$expect" \
+            || die_
+    fi
+    sleep 1
+    [  "0" = "$code" ] || exit_ $code # Quit on 1st failure
+}
+
+
+controller_()
+{
+    controller_app=$(realpath "${z_wave_stack_binaries_bin_dir}/ZW_zwave_ncp_serial_api_controller_"*"_REALTIME"*".elf" | head -n1)
+    file -E "${controller_app}"
+    ${controller_app} --pty
+}
+
+
+controller_cli_()
+{
+    while [ ! -e "${controller_log}" ] ; do sleep 1; done
+    screen -S "$name" -p 1 -t "controller" -X stuff "$@^M"
+    sleep 1
+    case $1 in
+        p)
+            PTY=$(grep 'PTY: ' ${controller_log} \
+                      | tail -n 1 | sed -e 's|PTY: \(.*\)|\1|g' )
+            ;;
+        n)
+            contid=$(grep 'NODE_ID: ' "${controller_log}" \
+                         | tail -n 1 | sed -e 's|NODE_ID: \(.*\)|\1|g' )
+            contid=$(printf "%04d" $contid)
+            ;;
+    esac
+    # TODO: print HOME_ID: from device: https://github.com/Z-Wave-Alliance/z-wave-stack/issues/732
+    while [ ! -e "${mqtt_log}" ] ; do sleep 1; done
+    [ "$homeid" != "" ] \
+        || homeid=$(sed -n -e 's|ucl/by-unid/zw-\(.*\)-\([0-9]*\)/.*|\1|gp' "$mqtt_log" | tail -n1)
+    contunid="zw-$homeid-$contid"
+}
+
+
+node_()
+{
+    node_app=$(realpath "${z_wave_stack_binaries_bin_dir}/ZW_zwave_soc_switch_on_off_"*"_REALTIME"*".elf" | head -n1)
+    [ "$debug" = "" ] || file -E "${node_app}"
+    ${node_app} --pty
+}
+
+
+node_cli_()
+{
+    while [ ! -e "${node_log}" ] ; do sleep 1; done
+    screen -S "$name" -p 2 -t "node" -X stuff "$@^M"
+    sleep 1
+    case $1 in
+        d)
+            DSK=$(grep 'DSK: ' "${node_log}" \
+                      | tail -n 1 | sed -e 's|DSK: \([0-9-]*\)|\1|g' )
+            SecurityCode=$(echo "$DSK" | sed -e 's|\([0-9]*\)-[0-9-]*$|\1|g')
+            ;;
+        n)
+            nodeid=$(grep 'NODE_ID: ' $node_log \
+                         | tail -n 1 | sed -e 's|NODE_ID: \(.*\)|\1|g' )
+            nodeid=$(printf "%04d" $nodeid)
+            ;;
+    esac
+    # TODO: print HOME_ID: from device: https://github.com/Z-Wave-Alliance/z-wave-stack/issues/732
+    while [ ! -e "${mqtt_log}" ] ; do sleep 1; done
+    [ "$homeid" != "" ] \
+        || homeid=$(sed -n -e 's|ucl/by-unid/zw-\(.*\)-\([0-9]*\)/.*|\1|gp' "$mqtt_log" | tail -n1)
+    nodeunid="zw-$homeid-$nodeid"
+}
+
+
+zpc_()
+{
+    controller_cli_ p
+    file -E "$PTY"
+    export ZPC_DEVICE=$(realpath -- "$PTY")
+    if [ "${ZPC_COMMAND}" = "/usr/bin/zpc" ] ; then
+        ZPC_COMMAND="${ZPC_COMMAND} --zpc.serial=${ZPC_DEVICE}"
+    fi
+    if [[ "${ZPC_COMMAND}" =~ $PWD/build/.*/zpc ]] ; then
+        ZPC_COMMAND="${ZPC_COMMAND} --zpc.serial=${ZPC_DEVICE}"
+    fi
+    ${ZPC_COMMAND}
+    sleep 1
+    zpc_cli_ help
+    die_
+
+}
+
+
+zpc_cli_()
+{
+    log_ "TODO: Fix console that eat some chars, and discard next workaround"
```

Comment:
Add reference to:

https://github.com/SiliconLabsSoftware/z-wave-engine-application-layer/issues/30